### ISSUE TITLE
introduction matrix.c is_sparse() function error

### ISSUE THREAD

**Issue by mennamorato** [lionleaf/parallel-c-programs#1]
Probably nobody is paying attention to this repo..i was going to try a pull request but its very old.
anyways... is_sparse does not work correctly because the formula is inverted, see here to fix it:
    
54   //if((matrix.cols * matrix.rows)/zero_elements >= sparse_threshold){  <-- BAD
55   if((zero_elements/(matrix.cols * matrix.rows)) >= sparse_threshold){
          return 1;
       }else{
         return 0;
       }

also to get rid of warnings about printf''ing pointers change the %d to %p here:
  186 // Attempting to multiply m and n, should work
  187 error = matrix_multiply(m,n,&p);
  188 printf("&p: %p\n",&p);
  189  printf("p: %p\n",p);

**Comment by lionleaf** [comment]
You're right, I didn't expect any pull requests on this. Nice find though!

It's an old repo with code I'm no longer using, but if you want to submit it as a pull request I'm happy to merge it
